ctru-rs/examples/ir-user-circle-pad-pro.rs

```diff
@@ -0,0 +1,259 @@
+//! A demo of using the ir:USER service to connect to the Circle Pad Pro.
+
+use ctru::prelude::*;
+use ctru::services::gfx::{Flush, Swap};
+use ctru::services::ir_user::{CirclePadProInputResponse, ConnectionStatus, IrDeviceId, IrUser};
+use ctru::services::svc::HandleExt;
+use ctru_sys::Handle;
+use std::time::Duration;
+
+const PACKET_INFO_SIZE: usize = 8;
+const MAX_PACKET_SIZE: usize = 32;
+const PACKET_COUNT: usize = 1;
+const PACKET_BUFFER_SIZE: usize = PACKET_COUNT * (PACKET_INFO_SIZE + MAX_PACKET_SIZE);
+const CPP_CONNECTION_POLLING_PERIOD_MS: u8 = 0x08;
+const CPP_POLLING_PERIOD_MS: u8 = 0x32;
```

Comment:
Nit: Might seem a bit too much, but you should probably specify in a comment that this configuration is specifically made for the CPP, and shouldn't be used in all IR use-cases.